packages/xod-client/src/editor/containers/Editor.jsx

```diff
@@ -47,26 +63,48 @@ class Editor extends React.Component {
       [COMMAND.SET_MODE_DEFAULT]: this.setModeDefault,
       [COMMAND.UNDO]: () => this.props.actions.undo(this.props.currentPatchPath),
       [COMMAND.REDO]: () => this.props.actions.redo(this.props.currentPatchPath),
+      [COMMAND.INSERT_NODE]: (event) => {
+        if (R.contains(event.target.type, ['textarea', 'input'])) return;
```

Comment:
Turns out we already had an util for this https://github.com/xodio/xod/blob/master/packages/xod-client/src/utils/browser.js#L19

Reply:
Thanks! I'll replace this one and couple of lines in the `Patch.jsx` :)